Mathlib/Geometry/Convex/Cone/Basic.lean

```diff
@@ -69,87 +67,74 @@ variable [Semiring R] [PartialOrder R] [IsOrderedRing R] [AddCommMonoid M]
 
-@[simp]
-theorem mem_mk {s : Set M} {h₁ h₂ x} : x ∈ mk (R := R) s h₁ h₂ ↔ x ∈ s :=
-  Iff.rfl
+@[simp] lemma mem_mk {h₁ h₂} : x ∈ mk (R := R) s h₁ h₂ ↔ x ∈ s := .rfl
 
 /-- Two `ConvexCone`s are equal if they have the same elements. -/
 @[ext]
-theorem ext {S T : ConvexCone R M} (h : ∀ x, x ∈ S ↔ x ∈ T) : S = T :=
-  SetLike.ext h
+theorem ext (h : ∀ x, x ∈ C₁ ↔ x ∈ C₂) : C₁ = C₂ := SetLike.ext h
 
+variable (C) in
 @[aesop safe apply (rule_sets := [SetLike])]
-theorem smul_mem {c : R} {x : M} (hc : 0 < c) (hx : x ∈ S) : c • x ∈ S :=
-  S.smul_mem' hc hx
+protected lemma smul_mem (hc : 0 < c) (hx : x ∈ C) : c • x ∈ C := C.smul_mem' hc hx
 
-theorem add_mem ⦃x⦄ (hx : x ∈ S) ⦃y⦄ (hy : y ∈ S) : x + y ∈ S :=
-  S.add_mem' hx hy
+variable (C) in
+protected lemma add_mem ⦃x⦄ (hx : x ∈ C) ⦃y⦄ (hy : y ∈ C) : x + y ∈ C := C.add_mem' hx hy
 
-instance : AddMemClass (ConvexCone R M) M where add_mem ha hb := add_mem _ ha hb
+instance : AddMemClass (ConvexCone R M) M where add_mem ha hb := add_mem' _ ha hb
 
 instance : Min (ConvexCone R M) :=
   ⟨fun S T =>
     ⟨S ∩ T, fun _ hc _ hx => ⟨S.smul_mem hc hx.1, T.smul_mem hc hx.2⟩, fun _ hx _ hy =>
       ⟨S.add_mem hx.1 hy.1, T.add_mem hx.2 hy.2⟩⟩⟩
 
-@[simp]
-theorem coe_inf : ((S ⊓ T : ConvexCone R M) : Set M) = ↑S ∩ ↑T :=
-  rfl
+variable (C₁ C₂) in
+@[simp, norm_cast] lemma coe_inf : (C₁ ⊓ C₂) = (C₁ ∩ C₂ : Set M) := rfl
 
-theorem mem_inf {x} : x ∈ S ⊓ T ↔ x ∈ S ∧ x ∈ T :=
-  Iff.rfl
+@[simp] lemma mem_inf : x ∈ C₁ ⊓ C₂ ↔ x ∈ C₁ ∧ x ∈ C₂ := .rfl
 
-instance : InfSet (ConvexCone R M) :=
-  ⟨fun S =>
-    ⟨⋂ s ∈ S, ↑s, fun _ hc _ hx => mem_biInter fun s hs => s.smul_mem hc <| mem_iInter₂.1 hx s hs,
-      fun _ hx _ hy =>
-      mem_biInter fun s hs => s.add_mem (mem_iInter₂.1 hx s hs) (mem_iInter₂.1 hy s hs)⟩⟩
+instance : InfSet (ConvexCone R M) where
+  sInf S :=
+    ⟨⋂ C ∈ S, C, fun _r hr _x hx ↦ mem_biInter fun C hC ↦ C.smul_mem hr <| mem_iInter₂.1 hx C hC,
+      fun _ hx _ hy ↦
+      mem_biInter fun C hC ↦ add_mem (mem_iInter₂.1 hx C hC) (mem_iInter₂.1 hy C hC)⟩
 
-@[simp]
-theorem coe_sInf (S : Set (ConvexCone R M)) : ↑(sInf S) = ⋂ s ∈ S, (s : Set M) :=
-  rfl
+@[simp, norm_cast]
+lemma coe_sInf (S : Set (ConvexCone R M)) : ↑(sInf S) = ⋂ C ∈ S, (C : Set M) := rfl
 
-theorem mem_sInf {x : M} {S : Set (ConvexCone R M)} : x ∈ sInf S ↔ ∀ s ∈ S, x ∈ s :=
-  mem_iInter₂
+@[simp] lemma mem_sInf {S : Set (ConvexCone R M)} : x ∈ sInf S ↔ ∀ C ∈ S, x ∈ C := mem_iInter₂
 
-@[simp]
+@[simp, norm_cast]
 theorem coe_iInf {ι : Sort*} (f : ι → ConvexCone R M) : ↑(iInf f) = ⋂ i, (f i : Set M) := by
   simp [iInf]
 
-theorem mem_iInf {ι : Sort*} {x : M} {f : ι → ConvexCone R M} : x ∈ iInf f ↔ ∀ i, x ∈ f i :=
+@[simp]
+lemma mem_iInf {ι : Sort*} {f : ι → ConvexCone R M} : x ∈ iInf f ↔ ∀ i, x ∈ f i :=
   mem_iInter₂.trans <| by simp
 
-variable (R)
-
 instance : Bot (ConvexCone R M) :=
   ⟨⟨∅, fun _ _ _ => False.elim, fun _ => False.elim⟩⟩
 
-theorem mem_bot (x : M) : (x ∈ (⊥ : ConvexCone R M)) = False :=
-  rfl
```

Comment:
This is clearly the wrong formulation of the assertion, but imo it's still useful to add a deprecation for it, since `rw [mem_bot]` is an occasionally helpful application